### ISSUE TITLE
Y047 false negative for conditionally defined aliases

### ISSUE THREAD

**Issue by AlexWaygood** [PyCQA/flake8-pyi#383]
This caused me some confusion in https://github.com/python/typeshed/pull/10177, where I _expected_ Y047 to be emitted, but it wasn't.

Code like this currently doesn't cause Y047 to be emitted, but it probably should:

```py
import sys
from typing_extensions import TypeAlias

if sys.version_info >= (3, 12):
    _Foo: TypeAlias = str
else:
    _Foo: TypeAlias = int
```

The reason is that this logic here is too naive:

https://github.com/PyCQA/flake8-pyi/blob/5b17698ba461b15c9d9fbfd5dee8b13c519bdce5/pyi.py#L1937-L1939

If an alias is conditionally defined, as above, the name of the type alias ("_Foo") will appear more than once in the file (it will appear once for each conditional definition), which leads flake8-pyi to assume that the type alias has been "used" at least once in the file that it's been defined in.
